libmultilabel/linear/estimator.py

```diff
@@ -0,0 +1,88 @@
+import re
+
+import scipy.sparse as sparse
+from sklearn.base import BaseEstimator
+from sklearn.utils.multiclass import unique_labels
+from sklearn.utils.validation import check_is_fitted
+
+import libmultilabel.linear as linear
+from libmultilabel.linear.utils import LINEAR_TECHNIQUES
+
+
+class MultiLabelMixin:
+    """Mixin class for multi-label classifiers.
+
+    Args:
+        metric_threshold (int, optional): The decision value threshold over which a label
+            is predicted as positive. Defaults to 0.
+        val_metric (str, optional): The evaluation metric for cross validation. Defaults to 'P@1'.
+    """
+
+    _estimator_type = 'classifier'
+
+    def __init__(self, metric_threshold=0, val_metric='P@1'):
+        self.metrics = None
+        self.metric_threshold = metric_threshold
+        self.val_metric = val_metric
+
+    def score(self, X: sparse.csr_matrix, y: sparse.csr_matrix):
+        """Sklearn estimator score method."""
+        if self.metrics is None:
+            self.metrics = linear.get_metrics(
+                self.metric_threshold,
+                [self.val_metric],
+                y.shape[1],
+            )
+        preds = self.predict(X)
+        target = y.toarray()
+        self.metrics.update(preds, target)
+        metric_dict = self.metrics.compute()
+        return metric_dict[self.val_metric]
+
+    def _more_tags(self):
+        return {'requires_y': True}
+
+
+class MultiLabelClassifier(MultiLabelMixin, BaseEstimator):
+    """Customized sklearn estimator for the multi-label classifier.
+
+    Args:
+        gridsearch (bool): Running grid search or not.
```

Comment:
This sentence reads as if the classifier is running grid search internally.
Perhaps something like "Whether the classifier is used in a grid search pipeline, see \<docs link\> for more."
On second thought, the parameter name might be misleading too.